solutions/client-landing-zone/client-folder/standard/applications-infrastructure/host-project/network/psc/google-apis/dns.yaml

```diff
@@ -0,0 +1,129 @@
+  ttl: 300
+  managedZoneRef:
+    name: host-project-id-internet-googleapis-dns # kpt-set: ${host-project-id}-internet-googleapis-dns
+  rrdatas:
+    - "10.255.255.254"
+---
+# Record Set for wildcard in googleapis.com DNS zone
+# SC-22
+apiVersion: dns.cnrm.cloud.google.com/v1beta1
+kind: DNSRecordSet
+metadata:
+  name: host-project-id-internet-googleapis-wildcard-rset # kpt-set: ${host-project-id}-internet-googleapis-wildcard-rset
+  namespace: client-name-networking # kpt-set: ${client-name}-networking
+  annotations:
+    cnrm.cloud.google.com/project-id: host-project-id # kpt-set: ${host-project-id}
+    config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/client-name-networking/ComputeNetwork/host-project-id-global-internet-vpc # kpt-set: compute.cnrm.cloud.google.com/namespaces/${client-name}-networking/ComputeNetwork/${host-project-id}-global-internet-vpc
+spec:
+  name: "*.googleapis.com."
+  type: "CNAME"
+  ttl: 300
+  managedZoneRef:
+    name: host-project-id-internet-googleapis-dns # kpt-set: ${host-project-id}-internet-googleapis-dns
+  rrdatas:
+    - "googleapis.com."
+---
+# Private DNS zone for gcr.io
+# SC-22
+apiVersion: dns.cnrm.cloud.google.com/v1beta1
+kind: DNSManagedZone
+metadata:
+  name: host-project-id-internet-gcrio-dns # kpt-set: ${host-project-id}-internet-gcrio-dns
+  namespace: client-name-networking # kpt-set: ${client-name}-networking
+  annotations:
+    cnrm.cloud.google.com/project-id: host-project-id # kpt-set: ${host-project-id}
+    config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/client-name-networking/ComputeNetwork/host-project-id-global-internet-vpc # kpt-set: compute.cnrm.cloud.google.com/namespaces/${client-name}-networking/ComputeNetwork/${host-project-id}-global-internet-vpc
+spec:
+  description: "Private DNS zone for gcr.io"
+  resourceID: internet-gcrio-dns
+  dnsName: "gcr.io."
+  visibility: private
+  privateVisibilityConfig:
+    networks:
+      - networkRef:
+          name: host-project-id-global-internet-vpc # kpt-set: ${host-project-id}-global-internet-vpc
+---
+# Record Set for gcr.io in gcr.io DNS zone
+# SC-22
+apiVersion: dns.cnrm.cloud.google.com/v1beta1
+kind: DNSRecordSet
+metadata:
+  name: host-project-id-internet-gcrio-rset # kpt-set: ${host-project-id}-internet-gcrio-rset
+  namespace: client-name-networking # kpt-set: ${client-name}-networking
+  annotations:
+    cnrm.cloud.google.com/project-id: host-project-id # kpt-set: ${host-project-id}
+    config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/client-name-networking/ComputeNetwork/host-project-id-global-internet-vpc # kpt-set: compute.cnrm.cloud.google.com/namespaces/${client-name}-networking/ComputeNetwork/${host-project-id}-global-internet-vpc
+spec:
+  name: "gcr.io."
+  type: "A"
+  ttl: 300
+  managedZoneRef:
+    name: host-project-id-internet-gcrio-dns # kpt-set: ${host-project-id}-internet-gcrio-dns
+  rrdatas:
+    - "10.255.255.254"
```

Comment:
you could replace this rrdatas to use
rrdatasRefs:
- kind: ComputeAddress
  name: host-project-id-internet-psc-apis-ip # kpt-set: ${host-project-id}-internet-psc-apis-ip

Reply:
updated, *however*, because the `solutions-validation` workflow is using the baked in, older, schemas of [kpt's kubeval function](https://catalog.kpt.dev/kubeval/v0.3/), it's failing with the following error:

`[error] dns.cnrm.cloud.google.com/v1beta1/DNSRecordSet/client1-networking/net-host-project-12345-internet-gcrio-rset rrdatas: rrdatas is required`

a few options comes to mind...
1. revert back to `spec.rrdatas` with a TODO to address once the workflow is fixed.
2. add `skip_kinds=DNSRecordSet` in the workflow's kubeval execution to ignore it for now (there's already a TODO in the workflow to update the schemas in the future).
3. update the workflow to use newer schemas (this will take longer to put in place)

for what it's worth, I'm leaning for option 2...

Reply:
yeah, I agree with the suggested solution